python/lsst/pipe/tasks/healSparseMapping.py

```diff
@@ -684,3 +687,232 @@ def _compute_nside_coverage_tract(self, tract_info):
         nside_coverage_tract = int(np.clip(nside_coverage_tract/2, 32, None))
 
         return nside_coverage_tract
+
+
+class ConsolidateHealSparsePropertyMapConnections(pipeBase.PipelineTaskConnections,
+                                                  dimensions=("band", "skymap",),
+                                                  defaultTemplates={"coaddName": "deep"}):
+    sky_map = pipeBase.connectionTypes.Input(
+        doc="Input definition of geometry/bbox and projection/wcs for coadded exposures",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+
+    # Create input/output connections for all possible maps defined in the
+    # registry.
+    for name in BasePropertyMap.registry:
+        vars()[f"{name}_map_min"] = pipeBase.connectionTypes.Input(
```

Comment:
Is this the preferred pattern for batch assignment of connections? It seems sensible on second thought but was unexpected to me at first.

Reply:
This is the only way that I know of to programatically make connections which must be class instance variables.  I've added additional comments, including a link to the magic stack overflow which told me how to do this: https://stackoverflow.com/questions/2519807/setting-a-class-attribute-with-a-given-name-in-python-while-defining-the-class